soroban-env-host/src/auth.rs

```diff
@@ -591,10 +591,12 @@ impl AuthorizationManager {
         }
         // No matching tracker found, hence the invocation isn't
         // authorized.
-        Err(HostError::from((
+        Err(host.err(
             ScErrorType::Auth,
             ScErrorCode::InvalidAction,
-        )))
+            "Not authorized",
```

Comment:
I think a better error message would be "Unauthorized function call for address", `[address]`. It would be nice to output the function itself, but that is a bit tricky, so can skip for now.